app/lib/actions/katello/content_view/publish.rb

```diff
@@ -102,7 +102,7 @@ def run
             end
           end
 
-          if ::Katello::CapsuleContent.sync_needed?(environment)
+          if ::Katello::CapsuleContent.sync_needed?(environment) && Setting[:foreman_proxy_content_auto_sync] == true
```

Comment:
the `== true` is implicit for testing booleans, so I'd drop it for better readability